### PR TITLE
Feature: 4 new breads, only craftable by the baker!

### PR DESCRIPTION
**Pull request by Mekle001**
# Changes proposed in this pull request:
- Sweet Bread
  - Slightly higher saturation than normal bread
  - Applies a speed boost & removes poison
  - Available at Baker level 3
- Add Milk-infused bread
  - Nutritionally, same as regular bread
  - Removes potion effects like Milk Bucket
  - If you're not hungry, you better hope you still have a milk-bucket
  - Available at Baker level 4
- Golden Bread
  - Nutritionally, same as regular bread
  - Immediately heals 2 hearts
  - Available at Baker level 5
- Chorus Bread
  - Higher saturation than normal bread
  - Teleports you to the surface in overworld-type dimensions
  - Requires new "Know the End" research @ university
- New "Know the End" research
  - Level 3 university, after researching stonemason
  - Requires a level 3 bakery & 64 chorus fruit
- End Stone recipe @ Stone Mason
  - Requires new "Know the End" research
- Fix a typo in carved pumpkin recipe

Review please

### CODE REVIEW

**Review by marchermans — CHANGES_REQUESTED** [review]
Some minor nitpicking:
A tooltip on the bread with an explanation (single line) of what its special features are would be nice.

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/ldtteam/minecolonies/pull-request/5970/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ldtteam/minecolonies)

**Review by marchermans — DISMISSED** [review]

File: src/main/java/com/minecolonies/coremod/items/ItemChorusBread.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.Style;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraft.world.World;
+import net.minecraft.world.dimension.DimensionType;
+import static com.minecolonies.api.util.constant.Constants.STACKSIZE;
+import java.util.List;
+
+/**
+ * Class handling Chorus Bread.
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/items/ItemChorusBread.java]: Not sure if I'd actually call this javadoc :D can you describe a little what the item is for here instead? same with other item classes

Comment (Mekle001) [reply src/main/java/com/minecolonies/coremod/items/ItemChorusBread.java]: Fixed

**Review by someaddons — APPROVED** [review]

**Pull request merged**